tests/semantic_analyser.cpp

```diff
@@ -496,29 +496,93 @@ kprobe:f / @mymap / { @mymap = "str" }
 
 TEST(semantic_analyser, ternary_expressions)
 {
-  test("kprobe:f { @x = pid < 10000 ? 1 : 2 }");
-  test(R"(kprobe:f { @x = pid < 10000 ? "lo" : "high" })");
+  // There are some supported types left out of this list
+  // as they don't make sense or cause other errors e.g.
+  // map aggregate functions and builtins
+  std::unordered_map<std::string, std::string> supported_types = {
+    { "1", "2" },
+    { "\"lo\"", "\"high\"" },
+    { "(\"hi\", 1)", "(\"bye\", 2)" },
+    { "printf(\"lo\")", "exit()" },
+    { "buf(\"mystr\", 5)", "buf(\"mystr\", 4)" },
+    { "macaddr(arg0)", "macaddr(arg1)" },
+    { "kstack(3)", "kstack(3)" },
+    { "ustack(3)", "ustack(3)" },
+    { "ntop(arg0)", "ntop(arg1)" },
+    { "nsecs(boot)", "nsecs(monotonic)" },
+    { "ksym(arg0)", "ksym(arg1)" },
+    { "usym(arg0)", "usym(arg1)" },
+    { "cgroup_path(1)", "cgroup_path(2)" },
+    { "strerror(1)", "strerror(2)" },
+  };
+
+  for (const auto &[left, right] : supported_types) {
+    test("kprobe:f { curtask ? " + left + " : " + right + " }");
+  }
+
   test("kprobe:f { pid < 10000 ? printf(\"lo\") : exit() }");
-  test("kprobe:f { curtask ? printf(\"lo\") : exit() }");
   test(R"(kprobe:f { @x = pid < 10000 ? printf("lo") : cat("/proc/uptime") })",
        3);
+  test("struct Foo { int x; } kprobe:f { curtask ? (struct Foo)*arg0 : (struct "
+       "Foo)*arg1 }",
+       1);
+  test("struct Foo { int x; } kprobe:f { curtask ? (struct Foo*)arg0 : (struct "
+       "Foo*)arg1 }");
+  test(
+      R"(kprobe:f { pid < 10000 ? ("a", "hellolongstr") : ("hellolongstr", "b") })");
```

Comment:
How is this test different from the `{ "\"lo\"", "\"high\"" },` variant above?

Reply:
It's testing tuple resizing (the `create_merged_tuple` function)